### ISSUE TITLE
A problem with Excel when there is a filtered date column

### ISSUE THREAD

**Issue by mail955** [NonFactors/MVC6.Grid#226]
Hi
When I have a filtered date column, and I export a list with the filtered column, the code does not refer to the date I requested.
This problem only happens when exporting to Excel, and only when filtering in a date type column. In the Grid view, it displays the filtered records properly, as well as whether the filter is in a number or text column.

Sample code:
```
            IGrid <TblTrans> grid = new Grid <TblTrans> (db.TblTrans);
            grid.ViewContext = new ViewContext {HttpContext = HttpContext};
            //grid.Query = new System.Collections.Specialized.NameValueCollection (Request.QueryString);
            // Request.QueryString;
            grid.Query = Request.QueryString;

            var r = grid.Rows;
            grid.Columns.Add (c => c.TransNow.Value.Date).Titled("Purchase Date"). Formatted ("{0: d}").AppendCss ("date"). Filterable (true);
            return await ExcelExporter.ExportAsync (grid);
```

Exporter Code:
```

        public static async Task<ActionResult> ExportAsync(IGrid grid, string fileName = "export.xlsx")
        {
            foreach (IGridColumn column in grid.Columns)
            {
                column.Filter.IsEnabled = true;
                column.Sort.IsEnabled = true;
            }


            // Using EPPlus from nuget
            //using (ExcelPackage package = new ExcelPackage())
            //{
                ExcelPackage package = new ExcelPackage(new FileInfo("export.xlsx"));
                Int32 row = 2;
                Int32 col = 1;

                package.Workbook.Worksheets.Add("Data");
                package.Workbook.Worksheets[1].View.RightToLeft = true;
                //IGrid grid = gridexport;// CreateExportableGrid(id);
                ExcelWorksheet sheet = package.Workbook.Worksheets["Data"];

                foreach (IGridColumn column in grid.Columns)
                {
                    sheet.Cells[1, col].Value = column.Title;
                    sheet.Column(col).Width = 18;
                    if (column.CssClasses == "date")
                    {
                        sheet.Column(col).Style.Numberformat.Format = "dd.MM.yyyy";
                    }
                    if (column.CssClasses == "time")
                    {
                        sheet.Column(col).Style.Numberformat.Format = "HH:mm";
                    }

                    col++;
                }

              // this grid.Rows is long time to go if date column filterd - this not filtered colmun
                foreach (IGridRow<object> gridRow in grid.Rows)
                {
                    col = 1;
                    foreach (IGridColumn column in grid.Columns)
                    {
                        sheet.Cells[row, col].Value = column.ValueFor(gridRow);

                        switch (column.CssClasses)
                        {
                            case "date":
                                sheet.Cells[row, col].Value = Convert.ToDateTime(sheet.Cells[row, col].Value.ToString());
                                sheet.Cells[row, col].Style.Numberformat.Format = "dd-MM-yy";

                                break;
                            case "time":
                                sheet.Cells[row, col].Value = Convert.ToDateTime(sheet.Cells[row, col].Value.ToString());
                                sheet.Cells[row, col].Style.Numberformat.Format = "HH:mm:ss";
                                break;
                            case "int":
                                if (sheet.Cells[row, col].Value.ToString() != "")
                                {
                                    sheet.Cells[row, col].Value = sheet.Cells[row, col].Value.ToString() == "" ? 0 : sheet.Cells[row, col].Value;
                                    if (Funcs.isNumeric(sheet.Cells[row, col].Value.ToString()))
                                    {
                                        sheet.Cells[row, col].Value = Convert.ToInt32(sheet.Cells[row, col].Value.ToString());

                                    }

                                }
                                break;
                            case "decimal":
                                sheet.Cells[row, col].Value = sheet.Cells[row, col].Value.ToString() == "" ? 0 : sheet.Cells[row, col].Value;
                                sheet.Cells[row, col].Value = Convert.ToDecimal(sheet.Cells[row, col].Value.ToString());

                                break;
                            default:
                                break;
                        }


                        col++;
                    }
                    row++;
                }
                FileContentResult file = new FileContentResult(package.GetAsByteArray(), "application/unknown");
                file.FileDownloadName = fileName;
                return file;
                //return new FileContentResult(package.GetAsByteArray(), "application/unknown", fileName);
            }

```
I thought maybe the problem was because I was moving the date in the DD / MM / YYYY format, so I also tried to move in the DD-MM-YYYY structure
In the 2 ways in the grid view he showed the correct data, but in actual export he removed all records.
It's about over a million records right now, and when you just want to get the last day's records out, it just pulls everything that's gone from the beginning of the activity

**Any help or idea what the problem might be will be greatly appreciated!
Thank you**

**Comment by Muchiachio** [comment]
Does your request query string have the same url as seen in the view?

**Comment by mail955** [comment]
yes.
my lang is hebrew.
But I don't know if it's related

Thanks

**Comment by Muchiachio** [comment]
Can you share a repro or try with your data on grid web repository?

**Comment by mail955** [comment]
Thanks :)
I'll do an example during the day and come up here
Thanks again :)

**Comment by mail955** [comment]
**ok
Thank you :)**

Finally, as I built the example project you requested, I realized what the problem was

Next line:
`grid.Columns.Add (c => c.TransNow.Value.Date) .Titled ("Purchase Date"). Formatted ("{0: d}"). AppendCss ("date"). Filterable (true);`

Because instead of a field:
`  c.TransNow`
  I wanted to bring only the date, without the hour, so I wrote it like this:
`c.TransNow.Value.Date`

The Grid SQL constructor probably recognizes that it is not a field in a table, but an expression, so it does not put it into SQL or build it.

I deleted
` .Value.Date`
And I just wrote
`c.TransNow`
And now everything is fine

thanks again :)
